x/checkers/keeper/keeper_integration_test.go

```diff
@@ -0,0 +1,108 @@
+package keeper_test
+
+import (
+	"testing"
+
+	checkersapp "github.com/b9lab/checkers/app"
+	"github.com/b9lab/checkers/x/checkers/keeper"
+	"github.com/b9lab/checkers/x/checkers/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+const (
+	transferEventCount            = 3 // As emitted by the bank
+	createEventCount              = 7
+	playEventCountFirst           = 8 // Extra "sender" attribute emitted by the bank
+	playEventCountNext            = 7
+	rejectEventCount              = 4
+	rejectEventCountWithTransfer  = 5 // Extra "sender" attribute emitted by the bank
+	forfeitEventCount             = 4
+	forfeitEventCountWithTransfer = 5 // Extra "sender" attribute emitted by the bank
+	alice                         = "cosmos1jmjfq0tplp9tmx4v9uemw72y4d2wa5nr3xn9d3"
+	bob                           = "cosmos1xyxs3skf3f4jfqeuv89yyaqvjc6lffavxqhc8g"
+	carol                         = "cosmos1e0w5t53nrq7p66fye6c8p0ynyhf6y24l4yuxd7"
+	balAlice                      = 50000000
+	balBob                        = 20000000
+	balCarol                      = 10000000
+)
+
+var (
+	checkersModuleAddress string
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	app         *checkersapp.App
+	msgServer   types.MsgServer
+	ctx         sdk.Context
+	queryClient types.QueryClient
+}
+
+func TestCheckersKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
+func (suite *IntegrationTestSuite) SetupTest() {
+	app := checkersapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+
+	app.AccountKeeper.SetParams(ctx, authtypes.DefaultParams())
+	app.BankKeeper.SetParams(ctx, banktypes.DefaultParams())
+	checkersModuleAddress = app.AccountKeeper.GetModuleAddress(types.ModuleName).String()
+
+	queryHelper := baseapp.NewQueryServerTestHelper(ctx, app.InterfaceRegistry())
+	types.RegisterQueryServer(queryHelper, app.CheckersKeeper)
+	queryClient := types.NewQueryClient(queryHelper)
+
+	suite.app = app
+	suite.msgServer = keeper.NewMsgServerImpl(app.CheckersKeeper)
+	suite.ctx = ctx
+	suite.queryClient = queryClient
+}
+
+func makeBalance(address string, balance int64) banktypes.Balance {
+	return banktypes.Balance{
+		Address: address,
+		Coins: sdk.Coins{
+			sdk.Coin{
+				Denom:  sdk.DefaultBondDenom,
+				Amount: sdk.NewInt(balance),
+			},
+		},
+	}
+}
+
+func getBankGenesis() *banktypes.GenesisState {
+	coins := []banktypes.Balance{
+		makeBalance(alice, balAlice),
+		makeBalance(bob, balBob),
+		makeBalance(carol, balCarol),
+	}
+	supply := banktypes.NewSupply(coins[0].Coins.Add(coins[1].Coins...).Add(coins[2].Coins...))
```

Comment:
I'm using cosmos-sdk v0.45.5. The `github.com/cosmos/cosmos-sdk/x/bank/types/Supply` was removed [here](https://github.com/cosmos/cosmos-sdk/commit/eb8aaf939513d3bf6e82bd13691cfbe9ef4ad28c#diff-7e91f6cf71ccad6269e2d5219ded9627667cf972c9a2d6ef7d2b7774bd8c009c) and [here](https://github.com/cosmos/cosmos-sdk/commit/eb8aaf939513d3bf6e82bd13691cfbe9ef4ad28c#diff-bea15b4c34f611322832ce50942a0ccbf7ef460548227795a504c3e72f6e85d4).
But as can be seen in the first reference `Supply.NewSupply` and `Supply.GetTotal` are just some kind of wrappers for Coins. The variable `supply` exists only in this scope and the bank module is initialiesed from the serialized genesis state.

To make the code working with Cosmos Sdk v0.45.5, I just modified it in the following way:
```
	totalSupplyCoins := balances[0].Coins.Add(balances[1].Coins...).Add(balances[2].Coins...)
	state := banktypes.NewGenesisState(
		banktypes.DefaultParams(),
		balances,
		totalSupplyCoins,
		[]banktypes.Metadata{})
```
PS: I am sure this is not the place for such feedback, but I did not find another.

Reply:
Also, do you know of a more elegant way of doing compared to `.Add().Add()`?